safekeeper/src/send_interpreted_wal.rs

```diff
@@ -430,7 +430,10 @@ impl InterpretedWalReader {
                         .with_context(|| "Failed to interpret WAL")?;
 
                         for (shard, record) in interpreted {
-                            if record.is_empty() {
+                            // Shard zero needs to track the start LSN of the latest record
+                            // in adition to the LSN of the next record to ingest. The former
+                            // is included in basebackup persisted by the compute in WAL.
```

Comment:
could you explain how we persist the basebackups in the WAL? I've thought the basebackup instead comes from the pageservers, and the computes request it.

Reply:
Yeah the comment is not great. Let's update it in a follow up to not waste CI time.

Basebackup itself is not persisted in the WAL. The `previous_lsn` (start LSN of the latest ingested record) is persisted in the WAL. It's not entirely clear to me where this happens. [This message](https://neondb.slack.com/archives/C08GTFBMV3K/p1741347252722569?thread_ts=1741276946.695059&cid=C08GTFBMV3K) from Arseny explains it
a little bit.